client/blockchain-service/src/handler.rs

```diff
@@ -1046,4 +1092,49 @@ impl BlockchainService {
         };
         current_tick == &last_tick_provided.saturating_add(provider_challenge_period)
     }
+
+    fn check_pending_forest_root_writes(&mut self) {
+        if let Some(mut rx) = self.forest_root_write_lock.take() {
+            // TODO: consider adding a timeout here to prevent a task from blocking the runtime.
+            match rx.try_recv() {
+                // If the channel is empty, means we still need to wait for the current task to finish.
+                Err(TryRecvError::Empty) => self.forest_root_write_lock = Some(rx),
+                _ => {}
+            }
+        }
+
+        // If we have a task writing to the runtime, we don't want to start another one.
+        if self.forest_root_write_lock.is_some() {
+            return;
+        }
+
+        let (tx, rx) = tokio::sync::oneshot::channel();
+        self.forest_root_write_lock = Some(rx);
```

Comment:
If there are no requests in the queue, what happens to this newly created lock? Does it get dropped at the end of of this function?